src/java/org/apache/cassandra/index/sai/utils/RangeIntersectionIterator.java

```diff
@@ -109,9 +107,9 @@ protected PrimaryKey computeNext()
 
     protected void performSkipTo(PrimaryKey nextToken)
     {
+        // (calling hasNext is a pessimisation, since it calls computeNext under the hood.  don't do it.)
```

Comment:
Pushed a clarification.  I think it's worth having a comment there instead of just a commit message, because adding the hasNext call "feels" like an optimization.

(There is still hasNext, inherited from AbstractIterator.)